modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3615,8 +3615,17 @@ void GDScriptAnalyzer::reduce_subscript(GDScriptParser::SubscriptNode *p_subscri
 			if (attr_type.is_set()) {
 				valid = true;
 				result_type = attr_type;
-				p_subscript->is_constant = p_subscript->attribute->is_constant;
-				p_subscript->reduced_value = p_subscript->attribute->reduced_value;
+				if (base_type.is_constant && result_type.kind == GDScriptParser::DataType::CLASS) {
```

Comment:
I wonder whether instead of doing this work here, it would be possible to simply call `reduce_identifier_from_base` and have this code as one of the special cases of that function. Would that make sense? Would it make it more reusable elsewhere?

Reply:
I successfully (for the first time, not that I had not tried) moved the code to `reduce_identifier_from_base()`!

Reply:
What about outer lookup?

Also instead of `p_base != nullptr && p_base->is_constant` you can say `base.is_constant`, no?

Reply:
> I successfully (for the first time, not that I had not tried) moved the code to `reduce_identifier_from_base()`!

It's easy to have opinions on someone else's code but I am glad it worked!